cluster/juju/layers/kubernetes-master/README.md

```diff
@@ -1,118 +1,102 @@
 
-You can add storage to the kubernetes charm in increments of 3 or greater:
+To attach a different resource to a deployed kubernetes-master charm, run the
+following command:
 
 ```
-juju add-storage kubernetes/0 disk-pool=ebs,3,1G
+juju attach kubernetes-master kubernetes=/path/to/kubernetes-master.tar.gz
 ```
 
-**Note**: Due to a limitation of raidz you can not add individual disks to an
-existing pool. Should you need to expand the storage of the raidz pool, the
-additional add-storage commands must be the same number of disks as the original
-command. At this point the charm will have two raidz pools added together, both
-of which could handle the  loss of one disk each.
+# Configuration
+
+This charm supports some configuration options to set up a Kubernetes cluster 
+that works in your environment:  
+
+#### dns_domain
 
-The storage code handles the addition of devices to the charm and when it
-recieves three disks creates a raidz pool that is mounted at the /srv/kubernetes
-directory by default. If you need the storage in another location you must
-change the `mount-point` value in layer.yaml before the charms is deployed.
+The domain name to use for the Kubernetes cluster for kube-dns.
 
-To avoid data loss you must attach the storage before making the connection to
-the etcd cluster.
+#### enable-dashboard-addons
 
-## State Events
-While this charm is meant to be a top layer, it can be used to build other
-solutions.  This charm sets or removes states from the reactive framework that
-other layers could react appropriately. The states that other layers would be
-interested in are as follows:
+Enables the installation of Kubernetes dashboard, Heapster, Grafana, and
+InfluxDB.
 
-**kubelet.available** - The hyperkube container has been run with the kubelet
-service and configuration that started the apiserver, controller-manager and
-scheduler containers.
+# DNS for the cluster
 
-**proxy.available** - The hyperkube container has been run with the proxy
-service and configuration that handles Kubernetes networking.
+The DNS add-on allows the pods to have a DNS names in addition to IP addresses.
+The Kubernetes cluster DNS server (based off the SkyDNS library) supports 
+forward lookups (A records), service lookups (SRV records) and reverse IP 
+address lookups (PTR records). More information about the DNS can be obtained
+from the [Kubernetes DNS admin guide](http://kubernetes.io/docs/admin/dns/).
 
-**kubectl.package.created** - Indicates the availability of the `kubectl`
-application along with the configuration needed to contact the cluster
-securely. You will need to download the `/home/ubuntu/kubectl_package.tar.gz`
-from the kubernetes leader unit to your machine so you can control the cluster.
+# Adding optional storage
 
-**skydns.available** - Indicates when the Domain Name System (DNS) for the
-cluster is operational.
+The kubernetes-master charm handles different persistent storage devices that
```

Comment:
The kubernetes-master is the only charm with the storage action. I feel if they are seeing errors with storage they may come to this readme for more information. The CDK bundle README does contain the instructions on how to deploy Ceph (which I agree is outside the scope of this charm's README). 

I just wanted to introduce the feature and give them the command to run because it is unique to this charm.